# Problem Statement - 

Consider yourself to be sam who is a data scientist. He is been approched by a medical issurance company to do data wrangling and find the meaning ful insight from it. 

Task to be performed :
    1. Start of the loading the 'issurance.csv' file and have look at the type of obejcts.

In [1]:
import pandas as pd

In [10]:
issarance = pd.read_csv("insurance (9).csv")

In [4]:
pwd

'C:\\Users\\Admin'

In [6]:
import os

In [7]:
path = "E:\Data Science With Python - Sakshi\Assignments\Assignment no 1"

In [9]:
os.chdir(path)

In [20]:
insurance = pd.read_csv("insurance (9).csv")

In [21]:
insurance

age     sex     bmi  children smoker     region      charges
0      19  female  27.900         0    yes  southwest  16884.92400
1      18    male  33.770         1     no  southeast   1725.55230
2      28    male  33.000         3     no  southeast   4449.46200
3      33    male  22.705         0     no  northwest  21984.47061
4      32    male  28.880         0     no  northwest   3866.85520
...   ...     ...     ...       ...    ...        ...          ...
1333   50    male  30.970         3     no  northwest  10600.54830
1334   18  female  31.920         0     no  northeast   2205.98080
1335   18  female  36.850         0     no  southeast   1629.83350
1336   21  female  25.800         0     no  southwest   2007.94500
1337   61  female  29.070         0    yes  northwest  29141.36030

[1338 rows x 7 columns]

In [22]:
insurance.head()

age     sex     bmi  children smoker     region      charges
0   19  female  27.900         0    yes  southwest  16884.92400
1   18    male  33.770         1     no  southeast   1725.55230
2   28    male  33.000         3     no  southeast   4449.46200
3   33    male  22.705         0     no  northwest  21984.47061
4   32    male  28.880         0     no  northwest   3866.85520

Task a.) Have a glance at the first 5 rows and the 5 column of the dataframe. After that find out the number of rows and column present in the datset

In [24]:
insurance.shape

(1338, 7)

Answer - There are 1338 rows and 7 colums

Task b.) Find the summary statistics on the dataset such as mean , median , standard deviation, count , minimum and maximum values.

In [23]:
insurance.describe()

age          bmi     children       charges
count  1338.000000  1338.000000  1338.000000   1338.000000
mean     39.207025    30.663397     1.094918  13270.422265
std      14.049960     6.098187     1.205493  12110.011237
min      18.000000    15.960000     0.000000   1121.873900
25%      27.000000    26.296250     0.000000   4740.287150
50%      39.000000    30.400000     1.000000   9382.033000
75%      51.000000    34.693750     2.000000  16639.912515
max      64.000000    53.130000     5.000000  63770.428010

In [29]:
insurance.mean()

age            39.207025
bmi            30.663397
children        1.094918
charges     13270.422265
dtype: float64

In [30]:
insurance.median()

age           39.000
bmi           30.400
children       1.000
charges     9382.033
dtype: float64

In [32]:
insurance.std()

age            14.049960
bmi             6.098187
children        1.205493
charges     12110.011237
dtype: float64

In [33]:
insurance.min()

age                18
sex            female
bmi             15.96
children            0
smoker             no
region      northeast
charges       1121.87
dtype: object

In [34]:
insurance.max()

age                64
sex              male
bmi             53.13
children            5
smoker            yes
region      southwest
charges       63770.4
dtype: object

Task c.) change the column name of "bmi" to Body_Mass_Index and find the correlation between "age", "children" and "charges"

In [37]:
#Answer = 
insurance.rename(columns={"bmi":"Body_Mass_Index"},inplace=True)

In [38]:
insurance

age     sex  Body_Mass_Index  children smoker     region      charges
0      19  female           27.900         0    yes  southwest  16884.92400
1      18    male           33.770         1     no  southeast   1725.55230
2      28    male           33.000         3     no  southeast   4449.46200
3      33    male           22.705         0     no  northwest  21984.47061
4      32    male           28.880         0     no  northwest   3866.85520
...   ...     ...              ...       ...    ...        ...          ...
1333   50    male           30.970         3     no  northwest  10600.54830
1334   18  female           31.920         0     no  northeast   2205.98080
1335   18  female           36.850         0     no  southeast   1629.83350
1336   21  female           25.800         0     no  southwest   2007.94500
1337   61  female           29.070         0    yes  northwest  29141.36030

[1338 rows x 7 columns]

In [39]:
insurance[['age','children','charges']].corr()

age  children   charges
age       1.000000  0.042469  0.299008
children  0.042469  1.000000  0.067998
charges   0.299008  0.067998  1.000000

Task d.) From entire 'insurance' dataframe, extract all the rows from number 500 to row 1000 and extract only 'sex', 'children' and'region' coulmns Store the final result in 'insurance_random_set'

In [41]:
insurance_random_set = insurance.loc[500:1000,['sex','children','region']]

In [42]:
insurance_random_set

sex  children     region
500     male         0  southwest
501     male         0  northeast
502     male         1  southeast
503     male         0  southeast
504   female         1  southeast
...      ...       ...        ...
996   female         3  southwest
997   female         0  southeast
998   female         3  northeast
999   female         0  northwest
1000    male         2  northwest

[501 rows x 3 columns]

In [44]:
#or

insurance_random_set1 = insurance.iloc[500:1000,[1,3,5]]

In [45]:
insurance_random_set1    #upper bond is excluded so instead of 1000 we need to put 1001

sex  children     region
500    male         0  southwest
501    male         0  northeast
502    male         1  southeast
503    male         0  southeast
504  female         1  southeast
..      ...       ...        ...
995  female         3  northeast
996  female         3  southwest
997  female         0  southeast
998  female         3  northeast
999  female         0  northwest

[500 rows x 3 columns]

In [46]:
#revised code

insurance_random_set2 = insurance.iloc[500:1001,[1,3,5]]

In [47]:
insurance_random_set2

sex  children     region
500     male         0  southwest
501     male         0  northeast
502     male         1  southeast
503     male         0  southeast
504   female         1  southeast
...      ...       ...        ...
996   female         3  southwest
997   female         0  southeast
998   female         3  northeast
999   female         0  northwest
1000    male         2  northwest

[501 rows x 3 columns]

Task e.) Sort the 'insurance' dataframe in the decending order of 'age'

In [49]:
insurance.sort_values(by = 'age') # by default it wil sort it by ascending

age     sex  Body_Mass_Index  children smoker     region      charges
1248   18  female           39.820         0     no  southeast   1633.96180
482    18  female           31.350         0     no  southeast   1622.18850
492    18  female           25.080         0     no  northeast   2196.47320
525    18  female           33.880         0     no  southeast  11482.63485
529    18    male           25.460         0     no  northeast   1708.00140
...   ...     ...              ...       ...    ...        ...          ...
398    64    male           25.600         2     no  southwest  14988.43200
335    64    male           34.500         0     no  southwest  13822.80300
378    64  female           30.115         3     no  northwest  16455.70785
1265   64    male           23.760         0    yes  southeast  26926.51440
635    64    male           38.190         0     no  northeast  14410.93210

[1338 rows x 7 columns]

In [51]:
insurance.sort_values(by='age', ascending = False)

age     sex  Body_Mass_Index  children smoker     region      charges
335    64    male           34.500         0     no  southwest  13822.80300
603    64  female           39.050         3     no  southeast  16085.12750
752    64    male           37.905         0     no  northwest  14210.53595
1265   64    male           23.760         0    yes  southeast  26926.51440
534    64    male           40.480         0     no  southeast  13831.11520
...   ...     ...              ...       ...    ...        ...          ...
942    18  female           40.185         0     no  northeast   2217.46915
46     18  female           38.665         2     no  northeast   3393.35635
295    18    male           22.990         0     no  northeast   1704.56810
50     18  female           35.625         0     no  northeast   2211.13075
648    18    male           28.500         0     no  northeast   1712.22700

[1338 rows x 7 columns]